sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs

```diff
@@ -88,6 +88,9 @@ void AddCustomerPolicies(HttpPipelinePosition position)
 
             var sanitizer = new HttpMessageSanitizer(diagnostics.LoggedHeaderNames.ToArray(), diagnostics.LoggedQueryParameters.ToArray());
 
+            responseClassifier ??= new ResponseClassifier();
+            responseClassifier.MessageSanitizer = sanitizer;
```

Comment:
We shouldn't be mutating an instance that is passed in by the customer.

Reply:
Addressed.